### ISSUE TITLE
problem while getting url_launcher package

### ISSUE THREAD

**Issue by MohammedAlNazili** [flutter/flutter#60435]
after    flutter pub get for url_launcher package 

```
Running Gradle task 'assembleDebug'...
Note: /home/malnozily-open-alt/my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/android/src/main/java/io/flutter/plugins/urllauncher/WebViewActivity.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
✓ Built build/app/outputs/apk/debug/app-debug.apk.
Installing build/app/outputs/apk/app.apk...
Debug service listening on ws://127.0.0.1:43551/SJA-MzRT9W8=/ws
Syncing files to device SM A205U...
```

then when we import the package and start using it :::

Compiler message:
```
../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:10:8: Error: Error when reading '../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_platform_interface-1.0.7/lib/url_launcher_platform_interface.dart': No such file or directory
import 'package:url_launcher_platform_interface/url_launcher_platform_interface.dart';
       ^
../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:88:29: Error: Getter not found: 'UrlLauncherPlatform'.
  final bool result = await UrlLauncherPlatform.instance.launch(
                            ^^^^^^^^^^^^^^^^^^^
../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:111:16: Error: Getter not found: 'UrlLauncherPlatform'.
  return await UrlLauncherPlatform.instance.canLaunch(urlString);
               ^^^^^^^^^^^^^^^^^^^
../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:126:16: Error: Getter not found: 'UrlLauncherPlatform'.
  return await UrlLauncherPlatform.instance.closeWebView();
               ^^^^^^^^^^^^^^^^^^^
```

**Comment by darshankawar** [comment]
Hi @MohammedAlNazili,
Can you please provide your `flutter doctor -v`, your `flutter run --verbose` and your `pubspec.yaml`. 
Thank you

**Comment by iLolz** [comment]
In iOS dispositive the url_Launcher can't open dialer, but in Android he open normally, someone have a success fix without call the method in native iOS?!

**Comment by MohammedAlNazili** [comment]
> Hi @MohammedAlNazili,
> Can you please provide your `flutter doctor -v`, your `flutter run --verbose` and your `pubspec.yaml`.
> Thank you

Sorry for late response  ^_^

flutter doctor -v

[✓] Flutter (Channel stable, v1.17.1, on Linux, locale en_GB.UTF-8)
    • Flutter version 1.17.1 at /home/malnozily-open-alt/my-development/flutter
    • Framework revision f7a6a7906b (8 weeks ago), 2020-05-12 18:39:00 -0700
    • Engine revision 6bc433c6b6
    • Dart version 2.8.2

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /home/malnozily-open-alt/Android/Sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /home/malnozily-open-alt/Desktop/My files/android-studio-ide-192.6392135-linux/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    ✗ Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions.

[✓] Android Studio (version 3.6)
    • Android Studio at /home/malnozily-open-alt/Desktop/My files/android-studio-ide-192.6392135-linux/android-studio
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.44.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.12.1

[✓] Connected device (1 available)
    • SM A205U • R58MA1RPQGL • android-arm64 • Android 9 (API 28)

! Doctor found issues in 1 category.


==========================================================================================================
flutter run --verbose


[  +16 ms] executing: [/home/malnozily-open-alt/my-development/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +23 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] f7a6a7906be96d2288f5d63a5a54c515a6e987fe
[        ] executing: [/home/malnozily-open-alt/my-development/flutter/] git tag --contains HEAD
[ +149 ms] Exit code 0 from: git tag --contains HEAD
[   +2 ms] 1.17.1
[  +11 ms] executing: [/home/malnozily-open-alt/my-development/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +5 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [/home/malnozily-open-alt/my-development/flutter/] git ls-remote --get-url origin
[   +3 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +99 ms] executing: [/home/malnozily-open-alt/my-development/flutter/] git rev-parse --abbrev-ref HEAD
[   +4 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[  +29 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +19 ms] executing: /home/malnozily-open-alt/Android/Sdk/platform-tools/adb devices -l
[  +45 ms] List of devices attached
           R58MA1RPQGL            device usb:3-6 product:a20psq model:SM_A205U device:a20p transport_id:2
[  +13 ms] /home/malnozily-open-alt/Android/Sdk/platform-tools/adb -s R58MA1RPQGL shell getprop
[ +111 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +145 ms] Found plugin url_launcher at /home/malnozily-open-alt/my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/
[   +8 ms] Found plugin url_launcher_macos at /home/malnozily-open-alt/my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_macos-0.0.1+7/
[   +5 ms] Found plugin url_launcher_web at /home/malnozily-open-alt/my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_web-0.1.2/
[  +80 ms] Found plugin url_launcher at /home/malnozily-open-alt/my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/
[   +2 ms] Found plugin url_launcher_macos at /home/malnozily-open-alt/my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_macos-0.0.1+7/
[   +3 ms] Found plugin url_launcher_web at /home/malnozily-open-alt/my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_web-0.1.2/
[  +42 ms] Generating /home/malnozily-open-alt/AndroidStudioProjects/flutter_app_url/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +24 ms] ro.hardware = exynos7885
[        ] ro.build.characteristics = phone
[  +38 ms] Launching lib/main.dart on SM A205U in debug mode...
[   +7 ms] /home/malnozily-open-alt/my-development/flutter/bin/cache/dart-sdk/bin/dart
/home/malnozily-open-alt/my-development/flutter/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root
/home/malnozily-open-alt/my-development/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names
-Ddart.developer.causal_async_stacks=true --output-dill /tmp/flutter_tool.ANITLR/app.dill --packages /home/malnozily-open-alt/AndroidStudioProjects/flutter_app_url/.packages
-Ddart.vm.profile=false -Ddart.vm.product=false
--bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions --enable-asserts
--track-widget-creation --filesystem-scheme org-dartlang-root --initialize-from-dill build/cache.dill
[  +22 ms] executing: /home/malnozily-open-alt/Android/Sdk/platform-tools/adb -s R58MA1RPQGL shell -x logcat -v time -t 1
[ +132 ms] Exit code 0 from: /home/malnozily-open-alt/Android/Sdk/platform-tools/adb -s R58MA1RPQGL shell -x logcat -v time -t 1
[        ] --------- beginning of system
           07-04 21:13:45.132 D/SDHMS:s ( 5770): SIOP:: AP:240(301,60) BAT:297(297,0) CHG:319(319,0) USB:0(0,0) LRP:321(321,-999) ATC:0(0,-999)
[  +18 ms] <- compile package:flutterappurl/main.dart
[  +12 ms] executing: /home/malnozily-open-alt/Android/Sdk/platform-tools/adb version
[   +6 ms] Android Debug Bridge version 1.0.41
           Version 30.0.3-6597393
           Installed as /home/malnozily-open-alt/Android/Sdk/platform-tools/adb
[   +2 ms] executing: /home/malnozily-open-alt/Android/Sdk/platform-tools/adb start-server
[   +7 ms] Building APK
[  +25 ms] Running Gradle task 'assembleDebug'...
[   +2 ms] gradle.properties already sets `android.enableR8`
[   +3 ms] Using gradle from /home/malnozily-open-alt/AndroidStudioProjects/flutter_app_url/android/gradlew.
[   +1 ms] /home/malnozily-open-alt/AndroidStudioProjects/flutter_app_url/android/gradlew mode: 33277 rwxrwxr-x.
[  +21 ms] executing: /home/malnozily-open-alt/Desktop/My files/android-studio-ide-192.6392135-linux/android-studio/jre/bin/java -version
[  +89 ms] Exit code 0 from: /home/malnozily-open-alt/Desktop/My files/android-studio-ide-192.6392135-linux/android-studio/jre/bin/java -version
[        ] openjdk version "1.8.0_212-release"
           OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
           OpenJDK 64-Bit Server VM (build 25.212-b4-5784211, mixed mode)
[   +4 ms] executing: [/home/malnozily-open-alt/AndroidStudioProjects/flutter_app_url/android/] /home/malnozily-open-alt/AndroidStudioProjects/flutter_app_url/android/gradlew
-Pverbose=true -Ptarget-platform=android-arm64 -Ptarget=/home/malnozily-open-alt/AndroidStudioProjects/flutter_app_url/lib/main.dart -Ptrack-widget-creation=true
-Pfilesystem-scheme=org-dartlang-root assembleDebug
[ +262 ms] 
                    Compiler message:
[   +2 ms] ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:10:8: Error: Error when reading
           '../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_platform_interface-1.0.7/lib/url_launcher_platform_interface.dart': No such file or
directory
[   +2 ms] import 'package:url_launcher_platform_interface/url_launcher_platform_interface.dart';
[  +10 ms]        ^
[+2694 ms] ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:88:29: Error: Getter not found: 'UrlLauncherPlatform'.
[        ]   final bool result = await UrlLauncherPlatform.instance.launch(
[        ]                             ^^^^^^^^^^^^^^^^^^^
[   +1 ms] ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:111:16: Error: Getter not found: 'UrlLauncherPlatform'.
[        ]   return await UrlLauncherPlatform.instance.canLaunch(urlString);
[        ]                ^^^^^^^^^^^^^^^^^^^
[        ] ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:126:16: Error: Getter not found: 'UrlLauncherPlatform'.
[        ]   return await UrlLauncherPlatform.instance.closeWebView();
[        ]                ^^^^^^^^^^^^^^^^^^^
[+1045 ms] > Task :app:compileFlutterBuildDebug
[   +1 ms] [  +21 ms] executing: [/home/malnozily-open-alt/my-development/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [  +36 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [        ] f7a6a7906be96d2288f5d63a5a54c515a6e987fe
[        ] [        ] executing: [/home/malnozily-open-alt/my-development/flutter/] git tag --contains HEAD
[        ] [ +192 ms] Exit code 0 from: git tag --contains HEAD
[        ] [   +1 ms] 1.17.1
[        ] [  +15 ms] executing: [/home/malnozily-open-alt/my-development/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[        ] [   +5 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] [        ] origin/stable
[        ] [        ] executing: [/home/malnozily-open-alt/my-development/flutter/] git ls-remote --get-url origin
[        ] [   +6 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] [        ] https://github.com/flutter/flutter.git
[        ] [  +99 ms] executing: [/home/malnozily-open-alt/my-development/flutter/] git rev-parse --abbrev-ref HEAD
[        ] [   +8 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] [        ] stable
[        ] [  +38 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [  +11 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[  +13 ms] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[  +13 ms] [        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +4 ms] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[   +4 ms] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +3 ms] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] [ +140 ms] Initializing file store
[   +2 ms] [  +13 ms] Done initializing file store
[        ] [  +43 ms] invalidated build due to missing files:
/home/malnozily-open-alt/my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_platform_interface-1.0.7/lib/url_launcher_platform_interface.dart
[ +523 ms] [+1450 ms] kernel_snapshot: Starting due to {InvalidatedReason.inputMissing}
[ +100 ms] [  +19 ms] /home/malnozily-open-alt/my-development/flutter/bin/cache/dart-sdk/bin/dart
/home/malnozily-open-alt/my-development/flutter/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root
/home/malnozily-open-alt/my-development/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --target=flutter -Ddart.developer.causal_async_stacks=true
-Ddart.vm.profile=false -Ddart.vm.product=false
--bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions --enable-asserts
--track-widget-creation --no-link-platform --packages /home/malnozily-open-alt/AndroidStudioProjects/flutter_app_url/.packages --output-dill
/home/malnozily-open-alt/AndroidStudioProjects/flutter_app_url/.dart_tool/flutter_build/ec52ba5a1921f2022156c52bb448169d/app.dill --depfile
/home/malnozily-open-alt/AndroidStudioProjects/flutter_app_url/.dart_tool/flutter_build/ec52ba5a1921f2022156c52bb448169d/kernel_snapshot.d package:flutterappurl/main.dart
[ +400 ms] [ +420 ms] 
[        ]            Compiler message:
[        ] [   +2 ms] ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:10:8: Error: Error when reading
'../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_platform_interface-1.0.7/lib/url_launcher_platform_interface.dart': No such file or directory
[   +2 ms] [        ] import 'package:url_launcher_platform_interface/url_launcher_platform_interface.dart';
[   +1 ms] [        ]        ^
[+2096 ms] [+2132 ms] ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:88:29: Error: Getter not found:
'UrlLauncherPlatform'.
[   +4 ms] [        ]   final bool result = await UrlLauncherPlatform.instance.launch(
[        ] [        ]                             ^^^^^^^^^^^^^^^^^^^
[   +3 ms] [   +3 ms] ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:111:16: Error: Getter not found:
'UrlLauncherPlatform'.
[        ] [        ]   return await UrlLauncherPlatform.instance.canLaunch(urlString);
[        ] [        ]                ^^^^^^^^^^^^^^^^^^^
[        ] [   +2 ms] ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:126:16: Error: Getter not found:
'UrlLauncherPlatform'.
[   +4 ms] [        ]   return await UrlLauncherPlatform.instance.closeWebView();
[        ] [        ]                ^^^^^^^^^^^^^^^^^^^
[+4486 ms] [+4467 ms] Persisting file store
[        ] [  +12 ms] Done persisting file store
[        ] [  +10 ms] "flutter assemble" took 8,801ms.
[        ] > Task :app:compileFlutterBuildDebug FAILED
[        ] [   +1 ms] Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
[   +5 ms] build failed.
[        ] #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
[        ] #1      AssembleCommand.runCommand (package:flutter_tools/src/commands/assemble.dart:202:7)
[        ] #2      _rootRunUnary (dart:async/zone.dart:1192:38)
[        ] #3      _CustomZone.runUnary (dart:async/zone.dart:1085:19)
[        ] #4      _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
[        ] #5      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
[        ] #6      Future._propagateToListeners (dart:async/future_impl.dart:711:32)
[        ] #7      Future._completeWithValue (dart:async/future_impl.dart:526:5)
[        ] #8      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:36:15)
[        ] #9      _completeOnAsyncReturn (dart:async-patch/async_patch.dart:298:13)
[        ] #10     BuildSystem.build (package:flutter_tools/src/build_system/build_system.dart)
[        ] #11     _rootRunUnary (dart:async/zone.dart:1192:38)
[        ] #12     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
[        ] #13     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
[        ] #14     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
[        ] #15     Future._propagateToListeners (dart:async/future_impl.dart:711:32)
[        ] #16     Future._completeWithValue (dart:async/future_impl.dart:526:5)
[        ] #17     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:36:15)
[        ] #18     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:298:13)
[        ] #19     _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart)
[        ] #20     _rootRunUnary (dart:async/zone.dart:1192:38)
[        ] #21     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
[        ] #22     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
[        ] #23     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
[        ] #24     Future._propagateToListeners (dart:async/future_impl.dart:711:32)
[   +5 ms] #25     Future._completeWithValue (dart:async/future_impl.dart:526:5)
[        ] #26     Future.wait.<anonymous closure> (dart:async/future.dart:402:22)
[        ] #27     _rootRunUnary (dart:async/zone.dart:1192:38)
[        ] #28     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
[        ] #29     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
[        ] #30     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
[        ] #31     Future._propagateToListeners (dart:async/future_impl.dart:711:32)
[        ] #32     Future._completeWithValue (dart:async/future_impl.dart:526:5)
[        ] #33     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:36:15)
[        ] #34     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:298:13)
[        ] #35     _BuildInstance._invokeInternal (package:flutter_tools/src/build_system/build_system.dart)
[        ] #36     _asyncErrorWrapperHelper.errorCallback (dart:async-patch/async_patch.dart:86:61)
[        ] #37     _rootRunBinary (dart:async/zone.dart:1204:38)
[        ] #38     _CustomZone.runBinary (dart:async/zone.dart:1093:19)
[        ] #39     _FutureListener.handleError (dart:async/future_impl.dart:155:20)
[        ] #40     Future._propagateToListeners.handleError (dart:async/future_impl.dart:694:47)
[        ] #41     Future._propagateToListeners (dart:async/future_impl.dart:715:24)
[        ] #42     Future._completeError (dart:async/future_impl.dart:534:5)
[        ] #43     _AsyncAwaitCompleter.completeError (dart:async-patch/async_patch.dart:43:15)
[        ] #44     KernelSnapshot.build (package:flutter_tools/src/build_system/targets/dart.dart)
[        ] #45     _rootRunUnary (dart:async/zone.dart:1192:38)
[        ] #46     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
[        ] #47     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
[        ] #48     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
[        ] #49     Future._propagateToListeners (dart:async/future_impl.dart:711:32)
[        ] #50     Future._completeWithValue (dart:async/future_impl.dart:526:5)
[        ] #51     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:36:15)
[        ] #52     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:298:13)
[        ] #53     KernelCompiler.compile (package:flutter_tools/src/compile.dart)
[        ] #54     _rootRunUnary (dart:async/zone.dart:1192:38)
[        ] #55     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
[        ] #56     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
[        ] #57     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
[        ] #58     Future._propagateToListeners (dart:async/future_impl.dart:711:32)
[        ] #59     Future._completeWithValue (dart:async/future_impl.dart:526:5)
[        ] #60     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:556:7)
[        ] #61     _rootRun (dart:async/zone.dart:1184:13)
[        ] #62     _CustomZone.run (dart:async/zone.dart:1077:19)
[        ] #63     _CustomZone.runGuarded (dart:async/zone.dart:979:7)
[        ] #64     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1019:23)
[        ] #65     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
[        ] #66     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
[        ] #67     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
[        ] #68     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
[   +1 ms] FAILURE: Build failed with an exception.
[        ] * Where:
[        ] Script '/home/malnozily-open-alt/my-development/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 882
[        ] * What went wrong:
[        ] Execution failed for task ':app:compileFlutterBuildDebug'.
[   +1 ms] > Process 'command '/home/malnozily-open-alt/my-development/flutter/bin/flutter'' finished with non-zero exit value 1
[        ] * Try:
[        ] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[        ] * Get more help at https://help.gradle.org
[        ] BUILD FAILED in 11s
[        ] 1 actionable task: 1 executed
[ +416 ms] Running Gradle task 'assembleDebug'... (completed in 12.3s)
[   +5 ms] Exception: Gradle task assembleDebug failed with exit code 1
[   +2 ms] "flutter run" took 13,127ms.

#0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
#1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:569:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:723:18)
#3      _rootRunUnary (dart:async/zone.dart:1192:38)
#4      _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#5      _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
#7      Future._propagateToListeners (dart:async/future_impl.dart:711:32)
#8      Future._completeWithValue (dart:async/future_impl.dart:526:5)
#9      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:36:15)
#10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:298:13)
#11     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
#12     _rootRunUnary (dart:async/zone.dart:1192:38)
#13     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#14     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#15     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
#16     Future._propagateToListeners (dart:async/future_impl.dart:711:32)
#17     Future._completeWithValue (dart:async/future_impl.dart:526:5)
#18     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:36:15)
#19     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:298:13)
#20     AndroidDevice.isLocalEmulator (package:flutter_tools/src/android/android_device.dart)
#21     _rootRunUnary (dart:async/zone.dart:1192:38)
#22     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#23     _FutureListener.handleValue (dart:async/future_impl.dart:141:18)
#24     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
#25     Future._propagateToListeners (dart:async/future_impl.dart:711:32)
#26     Future._completeWithValue (dart:async/future_impl.dart:526:5)
#27     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:556:7)
#28     _rootRun (dart:async/zone.dart:1184:13)
#29     _CustomZone.run (dart:async/zone.dart:1077:19)
#30     _CustomZone.runGuarded (dart:async/zone.dart:979:7)
#31     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1019:23)
#32     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
#33     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
#34     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#35     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)



==========================================================================
pubspec.yaml.


# The following line prevents the package from being accidentally published to
# pub.dev using `pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.0.0+1

environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  url_launcher: ^5.4.11


  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^0.1.3

dev_dependencies:
  flutter_test:
    sdk: flutter

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  # To add assets to your application, add an assets section, like this:
  # assets:
  #   - images/a_dot_burr.jpeg
  #   - images/a_dot_ham.jpeg

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  # fonts:
  #   - family: Schyler
  #     fonts:
  #       - asset: fonts/Schyler-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages


Thank you

**Comment by darshankawar** [comment]
Hi @MohammedAlNazili,

I tried the official [url_launcher](https://github.com/flutter/plugins/blob/master/packages/url_launcher/url_launcher/example/lib/main.dart) example and didn't see any error / exception after running `flutter pub get` and after launching the app. I was able to run the url properly without any issues.

I tested on latest dev channel (1.20.0-0.0.pre) and with latest `url_launcher` plugin on Android 8.

**Comment by iapicca** [comment]
Without additional information we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by zeeshanbhati** [comment]
> 
> 
> after flutter pub get for url_launcher package
> 
> ```
> Running Gradle task 'assembleDebug'...
> Note: /home/malnozily-open-alt/my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/android/src/main/java/io/flutter/plugins/urllauncher/WebViewActivity.java uses or overrides a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> ✓ Built build/app/outputs/apk/debug/app-debug.apk.
> Installing build/app/outputs/apk/app.apk...
> Debug service listening on ws://127.0.0.1:43551/SJA-MzRT9W8=/ws
> Syncing files to device SM A205U...
> ```
> 
> then when we import the package and start using it :::
> 
> Compiler message:
> 
> ```
> ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:10:8: Error: Error when reading '../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_platform_interface-1.0.7/lib/url_launcher_platform_interface.dart': No such file or directory
> import 'package:url_launcher_platform_interface/url_launcher_platform_interface.dart';
>        ^
> ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:88:29: Error: Getter not found: 'UrlLauncherPlatform'.
>   final bool result = await UrlLauncherPlatform.instance.launch(
>                             ^^^^^^^^^^^^^^^^^^^
> ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:111:16: Error: Getter not found: 'UrlLauncherPlatform'.
>   return await UrlLauncherPlatform.instance.canLaunch(urlString);
>                ^^^^^^^^^^^^^^^^^^^
> ../../my-development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.4.11/lib/url_launcher.dart:126:16: Error: Getter not found: 'UrlLauncherPlatform'.
>   return await UrlLauncherPlatform.instance.closeWebView();
>                ^^^^^^^^^^^^^^^^^^^
> ```

Did you find the Solution ?? I am facing the  same issue

**Comment by iapicca** [comment]
@zeeshanbhati 
this issue is closed, therefore is not actively worked on;
I recommend to follow the indication [above](https://github.com/flutter/flutter/issues/60435#issuecomment-665589267)

> Could everyone who still has this problem please file a new issue with the exact description of what happens, logs and the output of `flutter doctor -v`.
> All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by surajsahani** [comment]
I am getting the same error, although from mac everything works fine. but when I tried to deploy it from codemagic same error occurs. Any idea how can I solve it thank you in advance.
![Screenshot from 2021-05-17 13-43-56](https://user-images.githubusercontent.com/22853459/118454586-29009680-b716-11eb-80a2-94da9aa8db2a.png)

Here is full error code.

Error Message Via Codemagic
```
Failed to build iOS app
Error output from Xcode build:
↳
    ** ARCHIVE FAILED **


Xcode's output:
↳
    In file included from /Users/builder/programs/flutter_2_0_6/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.7.10/ios/Classes/FLTURLLauncherPlugin.m:7:
    /Users/builder/programs/flutter_2_0_6/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.7.10/ios/Classes/FLTURLLauncherPlugin.h:5:9: fatal error: 'Flutter/Flutter.h' file not found
    #import <Flutter/Flutter.h>
            ^~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete

Encountered error while archiveing for device.


Build failed :|
Failed to build for iOS
```

Flutter Doctor -v
![Screenshot from 2021-05-17 13-49-46](https://user-images.githubusercontent.com/22853459/118457581-847f5400-b717-11eb-9db1-ad42175764ea.png)

flutter run --verbose


 ```
flutter run --verbose
[ +123 ms] executing: uname -m
[  +73 ms] Exit code 0 from: uname -m
[        ] x86_64
[  +12 ms] executing: [/home/martial/Documents/sdks/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +7 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 06e2fd63574bad2edafbe4653104ed76871ee0b1
[   +1 ms] executing: [/home/martial/Documents/sdks/flutter/] git tag --points-at 06e2fd63574bad2edafbe4653104ed76871ee0b1
[  +20 ms] Exit code 0 from: git tag --points-at 06e2fd63574bad2edafbe4653104ed76871ee0b1
[        ] 2.2.0-10.3.pre
[  +13 ms] executing: [/home/martial/Documents/sdks/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +7 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/beta
[        ] executing: [/home/martial/Documents/sdks/flutter/] git ls-remote --get-url origin
[   +8 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[ +186 ms] executing: [/home/martial/Documents/sdks/flutter/] git rev-parse --abbrev-ref HEAD
[  +17 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] beta
[  +98 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +9 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +104 ms] executing: /home/martial/Android/Sdk/platform-tools/adb devices -l
[  +97 ms] List of devices attached
[   +9 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[  +55 ms] Downloading Web SDK...
[+1831 ms] Content https://storage.googleapis.com/flutter_infra_release/flutter/a123e75c6082da3a08f229b9c565e64b5b24a8a3/flutter-web-sdk-linux-x64.zip md5 hash: /Kj+exTS+YpDWV801r2JAw==
[+61600 ms] Downloading Web SDK... (completed in 63.4s)
[   +7 ms] executing: unzip -o -q /home/martial/Documents/sdks/flutter/bin/cache/downloads/storage.googleapis.com/flutter_infra_release/flutter/a123e75c6082da3a08f229b9c565e64b5b24a8a3/flutter-web-sdk-linux-x64.zip -d /home/martial/Documents/sdks/flutter/bin/cache/flutter_web_sdk
[+1499 ms] Exit code 0 from: unzip -o -q /home/martial/Documents/sdks/flutter/bin/cache/downloads/storage.googleapis.com/flutter_infra_release/flutter/a123e75c6082da3a08f229b9c565e64b5b24a8a3/flutter-web-sdk-linux-x64.zip -d /home/martial/Documents/sdks/flutter/bin/cache/flutter_web_sdk
[ +466 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +155 ms] No supported devices connected.
[  +67 ms] The following devices were found, but are not supported by this project:
           Chrome (web) • chrome • web-javascript • Google Chrome 90.0.4430.212
           If you would like your app to run on web, consider running `flutter create .` to generate projects for these platforms.
           
[  +18 ms] "flutter run" took 65,952ms.
[  +18 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.validateCommand (package:flutter_tools/src/commands/run.dart:453:7)
           <asynchronous suspension>
           #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1148:5)
           <asynchronous suspension>
           #3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1043:27)
           <asynchronous suspension>
           #4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #5      CommandRunner.runCommand (package:args/command_runner.dart:196:13)
           <asynchronous suspension>
           #6      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:284:9)
           <asynchronous suspension>
           #7      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #8      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:232:5)
           <asynchronous suspension>
           #9      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #10     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #11     main (package:flutter_tools/executable.dart:91:3)
           <asynchronous suspension>
           
           
[ +278 ms] ensureAnalyticsSent: 254ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
```

Pubspec.yaml files
![Screenshot from 2021-05-17 13-57-42](https://user-images.githubusercontent.com/22853459/118457924-dde78300-b717-11eb-9ad2-1c01a034370d.png)

**Comment by zeeshanbhati** [comment]
So In my case, the package was corrupted which was used by pubspec.yaml so the solution can be:
GO to this path in the flutter SDK 

As the error says that file is missing this could be a issue that package was not installed properly ,

Try to solve using these steps

 `\flutter\.pub-cache\hosted\pub.dartlang.org`
 All the different versions are available which you have downloaded now check , for the folder with version you are using in pubspec which si 5.7.10 in your case,  Check for the contents inside the folder if some of the files are missing then delete the folder and run `flutter pub get` again in your app. This should fix the error 

![ssss](https://user-images.githubusercontent.com/53969232/118460338-3ae43880-b71a-11eb-8409-e73755ba59f4.jpg)



> 
> 
> I am getting the same error, although from mac everything works fine. but when I tried to deploy it from codemagic same error occurs. Any idea how can I solve it thank you in advance.
> ![Screenshot from 2021-05-17 13-43-56](https://user-images.githubusercontent.com/22853459/118454586-29009680-b716-11eb-80a2-94da9aa8db2a.png)
> 
> Here is full error code.
> 
> Error Message Via Codemagic
> 
> ```
> Failed to build iOS app
> Error output from Xcode build:
> ↳
>     ** ARCHIVE FAILED **
> 
> 
> Xcode's output:
> ↳
>     In file included from /Users/builder/programs/flutter_2_0_6/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.7.10/ios/Classes/FLTURLLauncherPlugin.m:7:
>     /Users/builder/programs/flutter_2_0_6/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.7.10/ios/Classes/FLTURLLauncherPlugin.h:5:9: fatal error: 'Flutter/Flutter.h' file not found
>     #import <Flutter/Flutter.h>
>             ^~~~~~~~~~~~~~~~~~~
>     1 error generated.
>     note: Using new build system
>     note: Building targets in parallel
>     note: Planning build
>     note: Analyzing workspace
>     note: Constructing build description
>     note: Build preparation complete
> 
> Encountered error while archiveing for device.
> 
> 
> Build failed :|
> Failed to build for iOS
> ```
> 
> Flutter Doctor -v
> ![Screenshot from 2021-05-17 13-49-46](https://user-images.githubusercontent.com/22853459/118457581-847f5400-b717-11eb-9db1-ad42175764ea.png)
> 
> flutter run --verbose
> 
> ```
> flutter run --verbose
> [ +123 ms] executing: uname -m
> [  +73 ms] Exit code 0 from: uname -m
> [        ] x86_64
> [  +12 ms] executing: [/home/martial/Documents/sdks/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
> [   +7 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
> [        ] 06e2fd63574bad2edafbe4653104ed76871ee0b1
> [   +1 ms] executing: [/home/martial/Documents/sdks/flutter/] git tag --points-at 06e2fd63574bad2edafbe4653104ed76871ee0b1
> [  +20 ms] Exit code 0 from: git tag --points-at 06e2fd63574bad2edafbe4653104ed76871ee0b1
> [        ] 2.2.0-10.3.pre
> [  +13 ms] executing: [/home/martial/Documents/sdks/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
> [   +7 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
> [        ] origin/beta
> [        ] executing: [/home/martial/Documents/sdks/flutter/] git ls-remote --get-url origin
> [   +8 ms] Exit code 0 from: git ls-remote --get-url origin
> [        ] https://github.com/flutter/flutter.git
> [ +186 ms] executing: [/home/martial/Documents/sdks/flutter/] git rev-parse --abbrev-ref HEAD
> [  +17 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
> [        ] beta
> [  +98 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
> [   +9 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
> [ +104 ms] executing: /home/martial/Android/Sdk/platform-tools/adb devices -l
> [  +97 ms] List of devices attached
> [   +9 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
> [  +55 ms] Downloading Web SDK...
> [+1831 ms] Content https://storage.googleapis.com/flutter_infra_release/flutter/a123e75c6082da3a08f229b9c565e64b5b24a8a3/flutter-web-sdk-linux-x64.zip md5 hash: /Kj+exTS+YpDWV801r2JAw==
> [+61600 ms] Downloading Web SDK... (completed in 63.4s)
> [   +7 ms] executing: unzip -o -q /home/martial/Documents/sdks/flutter/bin/cache/downloads/storage.googleapis.com/flutter_infra_release/flutter/a123e75c6082da3a08f229b9c565e64b5b24a8a3/flutter-web-sdk-linux-x64.zip -d /home/martial/Documents/sdks/flutter/bin/cache/flutter_web_sdk
> [+1499 ms] Exit code 0 from: unzip -o -q /home/martial/Documents/sdks/flutter/bin/cache/downloads/storage.googleapis.com/flutter_infra_release/flutter/a123e75c6082da3a08f229b9c565e64b5b24a8a3/flutter-web-sdk-linux-x64.zip -d /home/martial/Documents/sdks/flutter/bin/cache/flutter_web_sdk
> [ +466 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
> [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
> [ +155 ms] No supported devices connected.
> [  +67 ms] The following devices were found, but are not supported by this project:
>           Chrome (web) • chrome • web-javascript • Google Chrome 90.0.4430.212
>           If you would like your app to run on web, consider running `flutter create .` to generate projects for these platforms.
>           
> [  +18 ms] "flutter run" took 65,952ms.
> [  +18 ms] 
>           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
>           #1      RunCommand.validateCommand (package:flutter_tools/src/commands/run.dart:453:7)
>           <asynchronous suspension>
>           #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1148:5)
>           <asynchronous suspension>
>           #3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1043:27)
>           <asynchronous suspension>
>           #4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
>           <asynchronous suspension>
>           #5      CommandRunner.runCommand (package:args/command_runner.dart:196:13)
>           <asynchronous suspension>
>           #6      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:284:9)
>           <asynchronous suspension>
>           #7      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
>           <asynchronous suspension>
>           #8      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:232:5)
>           <asynchronous suspension>
>           #9      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
>           <asynchronous suspension>
>           #10     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
>           <asynchronous suspension>
>           #11     main (package:flutter_tools/executable.dart:91:3)
>           <asynchronous suspension>
>           
>           
> [ +278 ms] ensureAnalyticsSent: 254ms
> [   +1 ms] Running shutdown hooks
> [        ] Shutdown hooks complete
> [        ] exiting with code 1
> ```
> 
> Pubspec.yaml files
> ![Screenshot from 2021-05-17 13-57-42](https://user-images.githubusercontent.com/22853459/118457924-dde78300-b717-11eb-9ad2-1c01a034370d.png)

**Comment by iapicca** [comment]
@surajsahani 
please follow the indication [above](https://github.com/flutter/flutter/issues/60435#issuecomment-665589267)

> Could everyone who still has this problem please file a new issue with the exact description of what happens, logs and the output of `flutter doctor -v`.
> All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by surajsahani** [comment]
@zeeshanbhati Thank for replying, how can i find above directory?

**Comment by zeeshanbhati** [comment]
> 
> 
> @zeeshanbhati Thank for replying, how can i find above directory?

Go to the location where flutter sdk is located inside that go to this path ` \.pub-cache\hosted\pub.dartlang.org`
